README.md

```diff
@@ -551,6 +552,24 @@ configs.maxMiddlewareStackExecutionAllowed = 3
 
 If an infinite loop is detected, mappersmith will throw an error.
 
+##### <a name="creating-middleware-optional-arguments-getFinalRequest"></a> getFinalRequest
+
+The `response` phase can optionally receive a function called "getFinalRequest". This function can be used to get access to the final request object (after the whole middleware chain has prepared and all `prepareRequest` been executed). This feature is useful in some scenarios, for example, when you want to get access to the request object without invoking `next`:
+
+```javascript
+const CircuitBreakerMiddleware = () => {
+  return () => ({
+    response(next, renew, getFinalRequest) {
```

Comment:
To be able to extend this further in the future, maybe the third argument here can be an object containing the utils?
```suggestion
    response(next, renew, { getFinalRequest }) {
```

Reply:
Definately looking over this in mappersmith v3